File: user-service/src/main/java/rs/raf/user_service/repository/UserRepository.java
package rs.raf.user_service.repository;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import rs.raf.user_service.entity.BaseUser;
import rs.raf.user_service.entity.Employee;

import java.util.Optional;

public interface UserRepository extends JpaRepository<BaseUser, Long> {
    Optional<BaseUser> findByEmail(String email);

    Page<BaseUser> findAll(Pageable pageable);  // Za paginaciju

    // Check if an employee exists with the given username
    boolean existsByUsername(String username);

    // Check if an employee exists with the given email
    boolean existsByEmail(String email);

    Optional<BaseUser> findByJmbg(String jmbg);
}

File: user-service-kotlin/src/main/kotlin/pack/userservicekotlin/repository/UserRepository.kt
package pack.userservicekotlin.repository

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import pack.userservicekotlin.domain.entities.BaseUser
import java.util.*

@Repository
interface UserRepository : JpaRepository<BaseUser, Long> {
    fun findByEmail(email: String): Optional<BaseUser>

    override fun findAll(pageable: Pageable): Page<BaseUser>

    fun existsByUsername(username: String): Boolean

    fun existsByEmail(email: String): Boolean

    fun findByJmbg(jmbg: String): Optional<BaseUser>
}
